extension/src/type_builder.rs

```diff
@@ -150,13 +150,15 @@ macro_rules! do_serialize {
         {
             let state = &*$state;
             let size = bincode::serialized_size(state)
-            .unwrap_or_else(|e| pgx::error!("serialization error {}", e));
+                .unwrap_or_else(|e| pgx::error!("serialization error {}", e)) + 1;
             let mut bytes = Vec::with_capacity(size as usize + 4);
             let mut varsize = [0; 4];
             unsafe {
                 pgx::set_varsize(&mut varsize as *mut _ as *mut _, size as _);
             }
             bytes.extend_from_slice(&varsize);
+            // serialization version; 1 for bincode is currently the only option
+            bytes.push(1);
```

Comment:
Why not make this a parameter of the function and have everything using it call with it?